src/main/java/org/broadinstitute/consent/http/service/VoteService.java

```diff
@@ -357,53 +361,44 @@ protected void notifyCustodiansOfApprovedDatasets(List<Dataset> datasets, User r
 
     // Find all the custodians, data owners, and data submitters to notify for each dataset
     datasets.forEach(d -> {
+      if (Objects.nonNull(d.getStudy())) {
+        Study study = d.getStudy();
+
+        // Data Submitter (study)
+        if (Objects.nonNull(study.getCreateUserId())) {
+          User submitter = userDAO.findUserById(study.getCreateUserId());
+          if (Objects.nonNull(submitter)) {
+            custodianMap.putIfAbsent(submitter, new HashSet<>());
+            custodianMap.get(submitter).add(d);
+          }
+        }
+
+        // Data Custodian (study)
+        if (Objects.nonNull(study.getProperties())) {
+          Set<StudyProperty> props = study.getProperties();
+          List<User> submitters = props.stream()
+              .filter(p -> p.getKey().equals("dataCustodianEmail"))
+              .map(p -> {
+                List<String> emailList = List.of(GsonUtil.getInstance().fromJson((String)p.getValue(), String[].class));
+                return userDAO.findUsersByEmailList(emailList);
+              }).flatMap(List::stream).toList();
+          if (!submitters.isEmpty()) {
+            submitters.forEach(s -> {
+              custodianMap.putIfAbsent(s, new HashSet<>());
+              custodianMap.get(s).add(d);
+            });
+          }
+        }
+      }
 
-      // Data Submitter
+      // Data Submitter (dataset)
       if (Objects.nonNull(d.getCreateUserId())) {
         User submitter = userDAO.findUserById(d.getCreateUserId());
         if (Objects.nonNull(submitter)) {
           custodianMap.putIfAbsent(submitter, new HashSet<>());
           custodianMap.get(submitter).add(d);
         }
       }
-
-      EmailValidator emailValidator = EmailValidator.getInstance();
```

Comment:
I think we still need to validate the emails with this class. It probably makes more sense to do that validation after all of the potential emails are collected up in custodianMap